nncf/torch/knowledge_distillation/knowledge_distillation_loss.py

```diff
@@ -92,6 +96,19 @@ def match_fn(obj):
 
         orig_model_loss_outputs = list(map(lambda x: x.getter(), filter(
             match_fn, orig_model_outputs_nested_obj_indexing.get_flat_nested_obj_indexing())))
+
+        if len(orig_model_loss_outputs) != len(compressed_model_loss_outputs):
+            nncf_logger.warning(f"KD: mismatch in the number of detected loss tensors in return value between original "
+                                f"and compressed models;\n"
+                                f"original has {len(orig_model_loss_outputs)} loss tensors,\n"
+                                f"compressed has {len(compressed_model_loss_outputs)}")
+        if not orig_model_loss_outputs:
+            nncf_logger.warning("KD: no loss outputs detected in original model, knowledge distillation not possible")
+            return None
+        if not compressed_model_loss_outputs:
+            nncf_logger.warning("KD: no loss outputs detected in compressed model, "
+                                "knowledge distillation not possible")
+            return None
```

Comment:
An improvement that I found useful not only when debugging this PR, but which could actually help users and future us to pinpoint what went wrong in the KD application.